tools/perf/ResultsComparer/src/ResultsComparer/VsProfiler/VsFunctionAllocationsComparer.cs

```diff
@@ -0,0 +1,72 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="VsFunctionAllocationsComparer.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using ResultsComparer.VsProfiler.Models;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace ResultsComparer.VsProfiler
+{
+    /// <summary>
+    /// An <see cref="Core.IResultsComparer"/> that can compare
+    /// Function Allocations reports generated by the VS .NET Object Allocation Tracker.
+    /// These reports are generated by selecting the desired rows in the VS UI, right-clicking
+    /// then copy-pasting the data into a text file.
+    /// </summary>
+    public class VsFunctionAllocationsComparer : VsProfilerComparer<VsProfilerFunctionAllocations>
+    {
+        /// <inheritdoc/>
+        public override string Name => "VS .NET Object Allocations - Functions";
+
+        /// <inheritdoc/>
+        protected override IDictionary<string, string> MetricNameMap => new Dictionary<string, string>()
+        {
+            { "Total", "Total Allocations" },
+            { "Self", "Self Allocations" },
+            { "Size", "Self Size (bytes)" },
+            { "Bytes", "Self Size (bytes)" }
+        };
+
+        /// <inheritdoc/>
+        protected override string DefaultMetric => "Self";
+
+        /// <inheritdoc/>
+        public override bool CanReadFile(string filePath)
+        {
+            using var reader = new StreamReader(filePath);
+            string firstLine = reader.ReadLine();
+            return firstLine != null
+                && firstLine.Contains("Name")
+                && (firstLine.Contains("Allocations") || firstLine.Contains("Bytes"));
+        }
+
+        /// <inheritdoc/>
+        protected override string GetItemId(VsProfilerFunctionAllocations item)
+        {
+            return item.Name;
+        }
+
+        /// <inheritdoc/>
+        protected override long GetMetricValue(VsProfilerFunctionAllocations item, string metric)
+        {
+            if (metric.Equals("Total", StringComparison.OrdinalIgnoreCase))

```

Comment:
Maybe to eliminate the overuse of `StringComparison.OrdinalIgnoreCase` we could rewrite the method as follows:
```c#
string metricToLower = metric.ToLower();

switch(metricToLower)
{
    case "total":
        return item.TotalAllocations;
    case "self":
        return item.SelfAllocations;
    // ...
}
```

Reply:
I was trying to avoid the `.ToLower()` call cause I assumed they would be some comments regarding the extra string allocation :)

Reply:
By the way, I don't remember where I read this but there was a suggestion to use `ToUpper...` rather than `ToLower...` methods in scenarios like these. Please verify

Reply:
My reasoning about use of switch was that it's fast as I understand

Reply:
It would be interesting to verify in this case whether the possible boost from using switch is for the cost of ToLower()/ToUpper(). I think a bigger flaw in this design is the fact that this method is called for each item/row in a report. Given that the same property is read for each item for the same metric, I think it would probably be more efficient to figure out which property to read once, then apply it to all items in the report, in that case I don't think the difference in ToUpper() vs switch would be significant. But I don't know what the impact of reflection would be. I think that's a discussion for another day.

According to this old post, ToLower() is slower than to ToUpper(): https://www.tabsoverspaces.com/233748-comparing-speed-of-toupper-toupperinvariant-tolower-and-tolowerinvariant-in-net-framework-and-net-core